### PR TITLE
fix: when lsp reference the file empty report error and render a hint “the file is empty”

### PR DESCRIPTION
**Pull request by jinzhongjia**
when use lsp reference to find function， if the file is empty, nvim will report error
图片

I added a piece of judgment logic to render the text the file is empty when the file is empty, and the highlight function will not be executed if the file is empty

图片

### CODE REVIEW

**Comment by jinzhongjia** [comment]
I haven't read the entire finder logic, so I don't know if there is redundancy in my diagnostic modification

File: lua/lspsaga/finder.lua [hunk 1]
```diff
@@ -626,7 +626,21 @@ function Finder:auto_open_preview()
   if not self.short_link[current_line] then
     return
   end
-  local content = self.short_link[current_line].preview or {}
+
+  local content
+  local table_length = 0
+  for key, value in pairs(self.short_link[current_line].preview) do
```
Comment (glepnir) [inline lua/lspsaga/finder.lua:632]: there can be `vim.tbl_count(self.short_link[current_line].preview`

Comment (jinzhongjia) [reply lua/lspsaga/finder.lua:632]: ok, i will change it

**Comment by jinzhongjia** [comment]
all right, i have changed it

**Pull request merged**